openedx/core/djangoapps/credit/tests/test_partition.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for In-Course Reverification Access Control Partition scheme
+"""
+
+import unittest
+
+from mock import Mock
+
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.credit.partition_schemes import (
+    VerificationPartitionScheme,
+    is_enrolled_in_verified_mode,
+    has_skipped_any_checkpoint,
+    has_completed_checkpoint,
+    was_denied_at_any_checkpoint
+)
+from student.models import CourseEnrollment
+
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from lms.djangoapps.verify_student.models import (
+    VerificationCheckpoint,
+    VerificationStatus,
+    SkippedReverification,
+)
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class ReverificationPartitionTest(ModuleStoreTestCase):
```

Comment:
I think some of these test cases might be a little bit too specific, which I think will make these tests difficult to maintain if we change the underlying implementation.

Would it be possible to generalize this test a bit?  I'm imagining something like:

1) Create N users, where each user is configured to be in a particular state (non-verified, verified, skipped, etc).
2) Call `get_group_for_user()` for each user and check that the user was put in the right group.

You could use `ddt` to parametrize the test, so you could re-use the test logic.

I think this is more maintainable because it allows us to change the implementation (how users are put into groups) without updating the tests.